src/powershell/FinOpsToolkit.psm1

```diff
@@ -151,39 +147,181 @@ function Get-FinOpsToolkitVersion
         $Preview
     )
 
+    $progress = $ProgressPreference
+    $ProgressPreference = 'SilentlyContinue'
     $releaseUri = 'https://api.github.com/repos/microsoft/cloud-hubs/releases'
-    [array]$releases = Invoke-WebRequest -Uri $releaseUri | ConvertFrom-Json | Where-Object { ($Preview) -or (-not $_.prerelease) }
+    
+    try
+    {
+        [array]$releases = Invoke-WebRequest -Uri $releaseUri | ConvertFrom-Json | Where-Object { ($Preview) -or (-not $_.prerelease) }
 
-    if ($Latest)
+        if ($Latest)
+        {
+            $releases = $releases | Select-Object -First 1
+            Write-Verbose -Message ($script:localizedData.FoundLatestRelease -f $releases.tag_name)
+        }
+
+        $output = @()
+        foreach ($release in $releases)
+        {
+            $properties = [ordered]@{
+                Name       = $release.name
+                PreRelease = $release.prerelease
+                Version    = $release.tag_name
+                Url        = $release.url
+                Assets     = @()
+            }
+
+            foreach ($asset in $release.assets)
+            {
+                $properties.Assets += @{
+                    Name = $asset.name
+                    Url  = $asset.browser_download_url
+                }
+            }
+
+            $output += New-Object -TypeName 'PSObject' -Property $properties
+        }
+
+        return $output
+    }
+    catch
     {
-        $releases = $releases | Select-Object -First 1
-        Write-Verbose -Message ($script:localizedData.FoundLatestRelease -f $releases.tag_name)
+        throw $_.Exception.Message
     }
+    finally
+    {
+        $ProgressPreference = $progress
+    }
+}
+
+<#
+    .SYNOPSIS
+        Deploys a FinOps hub instance.
+
+    .PARAMETER Name
+        Name of the FinOps hub instance.
+
+    .PARAMETER ResourceGroup
+        Name of the resource group to deploy to. Will be created if it doesn't exist.
+
+    .PARAMETER Location
+        Azure location to execute the deployment from.
+
+    .PARAMETER Version
+        Optional. Version of FinOps hub template to use. Defaults to latest.
 
-    $output = @()
-    foreach ($release in $releases)
+    .PARAMETER Preview
+        Optional. Deploy a pre-release version of FinOps Hub.
```

Comment:
I believe we usually say "Indicates..." for boolean values. Also adding in context about version.
```suggestion
        Optional. Indicates that a pre-release version of FinOps hub can be used when -Version is "latest".
```